common/src/main/kotlin/com/hartwig/actin/molecular/evidence/known/KnownEventResolver.kt

```diff
@@ -2,45 +2,73 @@ package com.hartwig.actin.molecular.evidence.known
 
 import com.hartwig.actin.datamodel.molecular.driver.CopyNumber
 import com.hartwig.actin.datamodel.molecular.driver.Disruption
+import com.hartwig.actin.datamodel.molecular.driver.GeneAlteration
 import com.hartwig.actin.datamodel.molecular.driver.HomozygousDisruption
+import com.hartwig.actin.datamodel.molecular.driver.VariantAlteration
 import com.hartwig.actin.molecular.evidence.matching.FusionMatchCriteria
 import com.hartwig.actin.molecular.evidence.matching.HotspotMatching
 import com.hartwig.actin.molecular.evidence.matching.RangeMatching
 import com.hartwig.actin.molecular.evidence.matching.VariantMatchCriteria
+import com.hartwig.actin.molecular.interpretation.GeneAlterationFactory
 import com.hartwig.serve.datamodel.molecular.KnownEvents
-import com.hartwig.serve.datamodel.molecular.common.GeneAlteration
 import com.hartwig.serve.datamodel.molecular.fusion.KnownFusion
 import com.hartwig.serve.datamodel.molecular.gene.KnownGene
 import com.hartwig.serve.datamodel.molecular.hotspot.KnownHotspot
 import com.hartwig.serve.datamodel.molecular.range.KnownCodon
 import com.hartwig.serve.datamodel.molecular.range.KnownExon
 
-class KnownEventResolver(private val knownEvents: KnownEvents, private val aggregatedKnownGenes: Set<KnownGene>) {
+class KnownEventResolver(
+    private val primaryKnownEvents: KnownEvents,
+    private val secondaryKnownEvents: KnownEvents,
+    private val aggregatedKnownGenes: Set<KnownGene>
+) {
 
-    fun resolveForVariant(variantMatchCriteria: VariantMatchCriteria): GeneAlteration? {
-        return findHotspot(knownEvents.hotspots(), variantMatchCriteria)
-            ?: findCodon(knownEvents.codons(), variantMatchCriteria)
-            ?: findExon(knownEvents.exons(), variantMatchCriteria)
+    fun resolveForVariant(variantMatchCriteria: VariantMatchCriteria): VariantAlteration {
+        val ckbAlteration = findHotspot(primaryKnownEvents.hotspots(), variantMatchCriteria)
+            ?: findCodon(primaryKnownEvents.codons(), variantMatchCriteria)
+            ?: findExon(primaryKnownEvents.exons(), variantMatchCriteria)
             ?: GeneLookup.find(aggregatedKnownGenes, variantMatchCriteria.gene)
+
+        val geneAlteration = GeneAlterationFactory.convertAlteration(variantMatchCriteria.gene, ckbAlteration)
+        val isHotspot =
+            HotspotFunctions.isHotspot(ckbAlteration) || findHotspot(secondaryKnownEvents.hotspots(), variantMatchCriteria) != null
```

Comment:
I am not sure how to improve this but in this line of code there are a lot of "hidden" assumptions. Key one being that the primary known event hotspots apparently are not all hotspots, while the secondary are. Also, you assume that the secondary hotspots never contain actual gene alteration information. 

I am fine with leaving as is, but we could maybe adjust `HotspotFunctions` to be used for every hotspot and then implement "if source contains primary -> check for protein effect, else return true". This way the function can be used for any hotspot